modules/custom/az_migration/src/Plugin/migrate/source/AZParagraphsItem.php

```diff
@@ -73,21 +73,24 @@ public function prepareRow(Row $row) {
         // Get Field API field values for each field collection item.
         $field_names = array_keys($this->getFields('field_collection_item', $field));
 
+        $field_collection_field_values = [];
         foreach ($field_names as $field_collection_field_name) {
-          $i = 0;
-          $field_collection_field_values = [];
-          foreach ($field_collection_data as $field_collection_data_item) {
-            $field_collection_value = $this->getFieldValues('field_collection_item',
-                                      $field_collection_field_name, $field_collection_data_item['value'],
-                                      $field_collection_data_item['revision_id']);
+          foreach ($field_collection_data as $delta => $field_collection_data_item) {
+            $field_collection_value = $this->getFieldValues(
+              'field_collection_item',
+              $field_collection_field_name,
+              $field_collection_data_item['value'],
+              $field_collection_data_item['revision_id']
+            );
             foreach ($field_collection_value as $field_collection_value_item) {
-              $field_collection_value_item['delta'] = $i;
-              $field_collection_field_values[] = $field_collection_value_item;
-              $i++;
+              $field_collection_field_values[$delta]['delta'] = $delta;
+              $field_collection_field_values[$delta][$field_collection_field_name][] = $field_collection_value_item;
             }
           }
-          $row->setSourceProperty($field_collection_field_name, $field_collection_field_values);
         }
+        $source_property_name = $field . '_values';
```

Comment:
```suggestion
        ksort($field_collection_field_values);
        $source_property_name = $field . '_values';
```

Reply:
For some reason it seems to matter more to the field plugin that the values be in order than the delta value. It seems to refuse to honor the delta value if it doesn't match the order the array is in...